### PR TITLE
Shot Handler pull request

### PR DESCRIPTION
**Pull request by xKoem**
Created ShotHanlder microservice with tests.

### CODE REVIEW

File: BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/shoot/PlayerShootCoordinate.java [hunk 1]
```diff
@@ -34,4 +35,19 @@ public String toString() {
...
+        return Objects.equals(player, that.player) &&
+                Objects.equals(coordinate, that.coordinate);
+    }
+
+    @Override
+    public int hashCode() {
+
+        return Objects.hash(player, coordinate);
+    }
+}
```
Comment (totine) [inline BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/shoot/PlayerShootCoordinate.java]: line

File: BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/shoot/PlayerShootResult.java [hunk 2]
```diff
@@ -33,4 +29,18 @@ public String toString() {
...
+        PlayerShootResult result = (PlayerShootResult) o;
+        return Objects.equals(player, result.player) &&
+                shootResult == result.shootResult;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(player, shootResult);
+    }
+}
```
Comment (totine) [inline BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/shoot/PlayerShootResult.java]: empty line

File: ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerFieldStatusEventListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        FieldStatusEvent event = new FieldStatusEvent(this, fieldStatus);
+        PlayerShootResult result = new PlayerShootResult(player, ShootResult.MISS);
+
+        eventListener.onApplicationEvent(event);
+
+        verify(queueService).sendObjectToQueue(Queues._15JudgePlayerShootResultQueue, result);
+    }
+
+
+    @Test(expectedExceptions = ClassCastException.class)
```
Comment (totine) [inline ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerFieldStatusEventListenerTest.java]: this test is to improve

File: ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerFieldStatusEventTest.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    @BeforeMethod
+    public void setUp() {
+        ShotHandlerQueueListener queueListener = mock(ShotHandlerQueueListener.class);
+        status = mock(FieldStatus.class);
+        event = new FieldStatusEvent(queueListener, status);
+    }
+
+    public void equalsTheSame_ReturnsTrue() {
+        assertEquals(event, event);
```
Comment (totine) [inline ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerFieldStatusEventTest.java]: :joy:

File: ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerShotActionEventListenerTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    public void sendingShotAction_SendsPlayerShootCoordinate_ToQueue12() {
+        ShotActionEvent event = new ShotActionEvent(this, action);
+        PlayerShootCoordinate playerShootCoordinate = new PlayerShootCoordinate(player, coordinate);
+        eventListener.onApplicationEvent(event);
+
+        verify(queueService).sendObjectToQueue(Queues._12BoardHandlerShotQueryQueue, playerShootCoordinate);
+    }
+
+    @Test(expectedExceptions = ClassCastException.class)
```
Comment (totine) [inline ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerShotActionEventListenerTest.java]: the same as above

File: ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerShotActionEventTest.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    @BeforeMethod
+    public void setUp() {
+        ShotHandlerQueueListener queueListener = mock(ShotHandlerQueueListener.class);
+        action = mock(PlayerAction.class);
+        event = new ShotActionEvent(queueListener, action);
+    }
+
+    public void equalsTheSame_ReturnsTrue() {
+        assertEquals(event, event);
```
Comment (totine) [inline ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerShotActionEventTest.java]: :joy: 2

File: BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/shoot/PlayerShootCoordinate.java [hunk 7]
```diff
@@ -34,4 +35,19 @@ public String toString() {
...
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        PlayerShootCoordinate that = (PlayerShootCoordinate) o;
+        return Objects.equals(player, that.player) &&
+                Objects.equals(coordinate, that.coordinate);
+    }
+
+    @Override
+    public int hashCode() {
+
```
[the remark this answers is not in the document]
Comment (boczkas) [reply BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/shoot/PlayerShootCoordinate.java]: :)

File: BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/shoot/PlayerShootCoordinate.java (hunk 1 above)
Comment (boczkas) [reply BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/shoot/PlayerShootCoordinate.java]: done

File: BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/shoot/PlayerShootResult.java (hunk 2 above)
Comment (boczkas) [reply BattleSheepLib/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/models/shoot/PlayerShootResult.java]: done

File: ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerFieldStatusEventListenerTest.java (hunk 3 above)
Comment (boczkas) [reply ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerFieldStatusEventListenerTest.java]: done

File: ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerFieldStatusEventTest.java (hunk 4 above)
Comment (boczkas) [reply ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerFieldStatusEventTest.java]: removed

File: ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerShotActionEventListenerTest.java (hunk 5 above)
Comment (boczkas) [reply ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerShotActionEventListenerTest.java]: refactored

File: ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerShotActionEventTest.java (hunk 6 above)
Comment (boczkas) [reply ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerShotActionEventTest.java]: removed

File: ShotHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/FieldStatusEvent.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
+package pl.jakubowskiprzemyslaw.tajgertim.event;
+
+import org.springframework.context.ApplicationEvent;
+import pl.jakubowskiprzemyslaw.tajgertim.models.coordinates.FieldStatus;
+
+import java.util.Objects;
+
+public class FieldStatusEvent extends ApplicationEvent {
```
Comment (gacikjakub) [inline ShotHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/FieldStatusEvent.java]: :+1:

File: ShotHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/FieldStatusEventListener.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public void onApplicationEvent(FieldStatusEvent event) {
+        FieldStatus fieldStatus = event.getFieldStatus();
+        FieldState fieldState = fieldStatus.getFieldState();
+
+        Player player = fieldStatus.getPlayer();
+        QueueObject playerShootResult;
+
+        switch (fieldState) {
```
Comment (gacikjakub) [inline ShotHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/FieldStatusEventListener.java]: suggest to extract to another private method

File: ShotHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/FieldStatusEventListener.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+import pl.jakubowskiprzemyslaw.tajgertim.models.coordinates.FieldState;
+import pl.jakubowskiprzemyslaw.tajgertim.models.coordinates.FieldStatus;
+import pl.jakubowskiprzemyslaw.tajgertim.models.player.Player;
+import pl.jakubowskiprzemyslaw.tajgertim.models.shoot.PlayerShootResult;
+import pl.jakubowskiprzemyslaw.tajgertim.models.shoot.ShootResult;
+import pl.jakubowskiprzemyslaw.tajgertim.queues.Queues;
+import pl.jakubowskiprzemyslaw.tajgertim.services.QueueService;
+
+@Component
+public class FieldStatusEventListener implements ApplicationListener<FieldStatusEvent> {
```
Comment (gacikjakub) [inline ShotHandler/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/event/FieldStatusEventListener.java]: :heart:

File: ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerFieldStatusEventTest.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+@Test
+public class ShotHandlerFieldStatusEventTest {
+
+    private FieldStatusEvent event;
+    private FieldStatus status;
+
+    @BeforeMethod
+    public void setUp() {
+        ShotHandlerQueueListener queueListener = mock(ShotHandlerQueueListener.class);
```
Comment (gacikjakub) [inline ShotHandler/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/ShotHandlerFieldStatusEventTest.java]: try to use MockitoAnnotations.initMocks(this); when you initialize many mocks. 
Remember about add @Mock annotation

**Pull request merged**